extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -221,7 +261,9 @@ public SecurityIdentity apply(Throwable throwable) {
 
                         String localRedirectUri = buildUri(context, isForceHttps(configContext), extraPath + extraQuery);
                         LOG.debugf("Local redirect URI: %s", localRedirectUri);
-                        return Uni.createFrom().failure(new AuthenticationRedirectException(localRedirectUri));
+                        return Uni.createFrom().failure(new AuthenticationRedirectException(
+                                shouldAutoRedirect(configContext, context),
+                                localRedirectUri));
```

Comment:
This constructor doesn't exist for AuthenticationRedirectException. The original code at 224 is the correct version.